### PR TITLE
Update sentry to v7.53.0

### PR DESCRIPTION
**Pull request by danjm**
## Explanation

This updates the sentry SDK packages we use.

This update in versions does come with breaking changes, but as far as I can tell, they don't affect our usage: https://github.com/getsentry/sentry-javascript/releases/tag/7.0.0

## Manual Testing Steps

Build the PR branch locally following the instructions in the Sentry section of https://github.com/MetaMask/metamask-extension/tree/develop/development#debugging-in-sentry

You should be able to see errors from the build of this branch on the sentry instance you created to follow those steps

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: test/e2e/tests/errors.spec.js [hunk 1]
```diff
@@ -5,7 +5,8 @@ const FixtureBuilder = require('../fixture-builder');
 describe('Sentry errors', function () {
   async function mockSentry(mockServer) {
     return await mockServer
-      .forPost('https://sentry.io/api/0000000/store/')
+      .forPost('https://sentry.io/api/0000000/envelope/')
```
Comment (danjm) [inline test/e2e/tests/errors.spec.js:8]: As discovered when manually testing, the error data is now sent as part of a POST request to the `/envelope` endpoint

File: test/e2e/tests/errors.spec.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ const FixtureBuilder = require('../fixture-builder');
...
+      .withBodyIncluding('Test Error')
```
Comment (danjm) [inline test/e2e/tests/errors.spec.js:9]: There are multiple requests sent to the `/envelope` endpoint, but only one is for the error event. This ensures that the function call only resolves once a request is made that includes the expected error data in the body.

File: test/e2e/tests/errors.spec.js [hunk 3]
```diff
@@ -48,7 +49,8 @@ describe('Sentry errors', function () {
           return isPending === false;
         }, 10000);
         const [mockedRequest] = await mockedEndpoint.getSeenRequests();
-        const mockJsonBody = mockedRequest.body.json;
+        const mockTextBody = mockedRequest.body.text.split('\n');
```
Comment (danjm) [inline test/e2e/tests/errors.spec.js:52]: It seems the body of the request is now a `string`, but one that is split into three parts.

**Comment by danjm** [comment]
I QA'd this following the steps here https://github.com/MetaMask/metamask-extension/tree/develop/development#sentry

I was able to successfully see an error in the sentry dashboard

**Comment by danjm** [comment]
@metamaskbot update-policies

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Comment by Gudahtt** [comment]
@metamaskbot update-policies

**Comment by Gudahtt** [comment]
It looks like that workflow is broken. I think I have found the problem, PR here: https://github.com/MetaMask/metamask-extension/pull/19774

**Comment by Gudahtt** [comment]
@metamaskbot update-policies

**Comment by Gudahtt** [comment]
OK, those policy changes look a lot more correct. Hopefully that means the bug is fixed.

**Comment by Gudahtt** [comment]
@SocketSecurity ignore @sentry/browser@7.56.0

This package is intended to make network calls; network access does not pose a risk.

**Review by Gudahtt — APPROVED** [review]

**Review by PeterYinusa — APPROVED** [review]
LGTM
Verified sentry events flow through to the error monitoring dashboard

**Pull request merged**